### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by Zelalem1222**
## In this PR 

- Icreated a method called current_user in ApplicationController to make current user data available to all controllers.
        - It will return the first user from the database.

- I created forms to perform the following functions:
        - Creates a Post on behalf of the current_user (using the method  created in  ApplicationController).
        - Create a comment on behalf of the current_user (using the method created in  ApplicationController).

- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+      redirect_to new_user_post_path(@user.id)
+      flash[:notice] = 'An error has occurred while creating the post'
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:tittle, :text, :comments_counter, :likes_counter)
+  end
```
Comment (omar-muhamad) [inline app/controllers/posts_controller.rb]: - [x] Kindly only permit input params, in this case for post you only allow `:title` and `:text`. 
`comments_counter` and `likes_counter` suppose to be auto-updated through the post model.

File: config/routes.rb [hunk 2]
```diff
@@ -6,6 +6,9 @@
    root to: "users#index"
 
    resources :users, only: [:index, :show] do
-      resources :posts, only: [:index, :show]
+      resources :posts, only: [:index, :show , :new, :create] do
+        resources :comments, only: [:create]
```
Comment (omar-muhamad) [inline config/routes.rb]: - [x] Nice work for the routes here, but please add a route for the comment `new` method as you added for `create`.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,9 +1,34 @@
...
+
+    if @post.save
+      @post.update_user_post_counter(params[:user_id])
+      redirect_to user_post_path(@user.id)
+      flash[:notice] = 'Your comment was successfully created'
+    else
+      redirect_to new_user_post_path(@user.id)
+      flash[:notice] = 'An error has occurred while creating the post'
+    end
+  end
```
Comment (omar-muhamad) [inline app/controllers/posts_controller.rb]: - [x] While creating a new post it raises an error in the image below, So please fix that.

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Zelalem1222,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by KayLemba — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @Zelalem1222  , 

Great work :+1: 
The app successfully updates the local storage when I mark a todo item as completed :clap: 
Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**